packages/host/app/components/ai-assistant/formatted-message.gts

```diff
@@ -109,6 +121,60 @@ export default class FormattedMessage extends Component<FormattedMessageSignatur
     return index === this.htmlGroups.length - 1;
   };
 
+  private createCodePatchAction = (codeData: CodeData) => {
+    let codePatchAction = new CodePatchAction(getOwner(this)!, codeData);
```

Comment:
CodePatchAction is a an object that holds the state of the code patch function (ready, applied, applying, failed) and has the functionality to apply the match.

The owner is passed and set in `CodePatchAction` because the owner is needed in order to inject services, such as loader service, command service, card service.

Reply:
We should consider keeping this somewhere besides component state so that if a user closes and reopens the AI Assistant, the state is correct.

Reply:
I would defer this to the commands integration work (where we will persist historic state of the code patch applied commands). I'll make a note in that ticket.